### ISSUE TITLE
runtime error with ghcjs

### ISSUE THREAD

**Issue by jhegedus42** [plow-technologies/ghcjs-servant-client#1]
<img width="1374" alt="screen shot 2015-10-11 at 15 40 05" src="https://cloud.githubusercontent.com/assets/6671476/10416838/ccabae8e-702f-11e5-851d-2dc1af666a6d.png">
<img width="1680" alt="screen shot 2015-10-11 at 15 40 30" src="https://cloud.githubusercontent.com/assets/6671476/10416841/e4f954e6-702f-11e5-8329-f7470417a073.png">

**Comment by jhegedus42** [comment]
any suggestions what might be causing this? should this library work with ghcjs ?

**Comment by smurphy8** [comment]
I'll look more closely later bit usually I get that when there is a
javascript routine that does not return

**Comment by jhegedus42** [comment]
Many thanks. Luite said on IRC that this error happens probably because of changes in GHCJS.  With  what version of GHCJS do you use the "new_branch" -branch of ghjcs-servant-client ? 

I am trying to revert to an older version of GHCJS to see if I can get this simple example working with ghcjs-servant-client. If you could tell me the hash of the GHCJS for which you know that works with ghcjs-servant-client ( "new_branch"-branch ) for your programs then knowing this hash would save me a lot of trial and error :)  

So If I can get this example to work with an earlier version of GHCJS then that means that the problem is not with ghcjs-servant-client but somewhere in GHCJS.

**Comment by luite** [comment]
Yes, something changed in the action that's being run for the entry point of the program. It used to be something that did less exception handling by itself, and wasn't always available, leading to other problems.

The new version sometimes gets stuck in an infinite loop apparently. I'll look into this. I don't know if there's any specific precondition that makes it happen, perhaps it's very simple. If anyone has a short program to reproduce this, without external (not built by `ghcjs-boot`) dependencies, that would probably help me find it more quickly.

this is the fix, reverting this should be a workaround for this issue, but causes another one: https://github.com/ghcjs/ghcjs/commit/ef9c97937c9eb58fff66884fe9b7df788cd671cb

**Comment by smurphy8** [comment]
New branch should be used .  It was working with improved base as of Friday.

Sorry for the naming of the branch.  We'll  get it fixed up.

**Comment by jhegedus42** [comment]
I try to make an error reproducing, servant independent code, but I might not be very fast at that... but I'll try.

**Comment by jhegedus42** [comment]
Here is an example of the crashing app : http://www.viizio.com/~joco/test.jsexe/

**Comment by jhegedus42** [comment]
hey Guys, here is a repo that - when compiled with latest ghcjs- gives the error : https://github.com/jhegedus42/reproduce-ghcjs-servant-client-bug

i was trying to create a minimalistic example by merging the servant api dependency into a single God-file but I think succeeding with that would take me a lot of time... maybe some of you guys can figure out something clever in 5 minutes on how to reproduce this bug in 5 lines of code... i have to admit that i am not such a great expert in haskell/servant/ghcjs yet ... i can do simple things but here i think i hit a hard wall - hope that this repo might give some ideas to you guys on how to create a simple executable that reproduces this error...

**Comment by smurphy8** [comment]
I guess we weren't clever enough @jhegedus42 sorry this slipped away but we are not actively working on this package anymore.
